packages/transformer/src/IModelImporter.ts

```diff
@@ -207,7 +235,19 @@ export class IModelImporter implements Required<IModelImportOptions> {
       }
     } else {
       if (undefined !== elementProps.id) {
-        this.onUpdateElement(elementProps);
+        try {
+          this.onUpdateElement(elementProps);
+        } catch(err) {
```

Comment:
Can we hoist this catch up into the calling function in IModelTransformer, and move the entire duplicate code map into the transformer?

That way it should be easier to keep that map as private, it really shouldn't be public

Reply:
I don't think hoisting it up would be correct because `importElement` might call `onInsertElement`. We need to react to `DuplicateCode` error only when it comes from `onUpdateElement`. It might hide valid duplicate code cases and these will be impossible to resolve after the transformation.

I made the map public in case the derived transformer classes needed the temporary state of the element, but since there are no such cases yet I can make it private.

Reply:
I now noticed you've added `importer.finalize`, I think that is a much better way of doing it :) that looks good to me. Making the map private is sufficient then, thanks!